### ISSUE TITLE
Dropping EventLoopHandle after running a request causes the program to hang.

### ISSUE THREAD

**Issue by abrodersen** [tomusdrw/rust-web3#80]
Dropping `EventLoopHandle` before dropping `Web3<T>` causes a hang if at least one request has been issued. Example code:

````
extern crate web3;

use web3::futures::Future;

fn main() {
  let (eloop, http) = web3::transports::Http::new("http://localhost:8545").unwrap();
  let web3 = web3::Web3::new(http);
  let _accounts = web3.eth().accounts().wait().unwrap();

  drop(eloop);
}
````

This became an issue when I created a struct that owned both `EventLoopHandle` and `Web3<T>`. If my struct fields were in the wrong order, the program would hang when attempting to drop my struct. If I switched the order of the fields, the program would run to completion.

**Comment by tomusdrw** [comment]
Good catch it's really interesting why it fails.

**Comment by aleksuss** [comment]
I have the same issue

**Comment by aleksuss** [comment]
FYI: But if I use an event loop from tokio everything is OK

**Comment by yangby-cryptape** [comment]
After test, I think it is hang at [`EventLoopHandle::Drop`](https://github.com/tomusdrw/rust-web3/blob/626620b/src/transports/shared.rs#L75).

**Comment by yangby-cryptape** [comment]
Finally, I have known what happened.

It hangs at [`event_loop.turn(None);`](https://github.com/tomusdrw/rust-web3/blob/626620b/src/transports/shared.rs#L47), **before checking** the [`done`](https://github.com/tomusdrw/rust-web3/blob/626620b/src/transports/shared.rs#L46) flag.

When we want to shutdown the `event_loop`, we stop send any new message, and set the done flag.
But the `event_loop` will check the done flag **after next message**, so it hangs **at waiting for next message**.
So, it hangs.

The simplest way (but stupid?) to fix this bug is: after set the done flag, send to it a new message.

**Comment by tomusdrw** [comment]
Perhaps we could supply a timeout to turn?

**Comment by synlestidae** [comment]
@yangby-cryptape is correct I think. I added `self.remote.spawn(|_| Ok(()))` before the thread join. It seems to fix it on my end. 

#144

**Comment by tomusdrw** [comment]
Resolved via #144
